docs/architecture/adr-008-app-caller-cbs/adr-008-app-caller-cbs.md

```diff
@@ -229,6 +227,21 @@ func OnRecvPacket(
         // deduct consumed gas from original context
         ctx = ctx.WithGasLimit(ctx.GasMeter().RemainingGas() - cbCtx.GasMeter().GasConsumed())
         if err != nil {
+            // NOTE: by returning an error acknowledgement, it is assumed that the
+            // base IBC application on the counterparty callback stack will be able
+            // to properly unmarshal the error acknowledgement. It should not expect
+            // some custom error acknowledgement. If it does, failed acknowledgements
+            // will be unsuccessfully processed which can be catastrophic in processing
+            // refund logic. 
+            // 
+            // If this issue is a serious concern, an ADR 8 implementation can construct its own
+            // acknowledgement type which wraps the underlying application acknowledgement. This 
+            // would require deployment on both sides of the packet flow, in addition to version 
+            // negotiation to enable the custom acknowledgement type usage. 
+            // 
+            // Future improvmenets should allow for each IBC application in a stack of
+            // callbacks to provide their own acknowledgement without disrupting the unmarshaling
+            // of an application above or below it in the stack. 
```

Comment:
I think this is a serious concern, but I think for practical purposes it's okay to move forward with some assumption which is partially unsafe. Improvements to the middleware design should fix this issue. I thought it best to leave a lot of documentation here. It might be nice to double check no base applications are using custom failed acks